index/scorch/persister.go

```diff
@@ -145,22 +145,16 @@ OUTER:
 	}
 }
 
-func (s *Scorch) persistSnapshot(snapshot *IndexSnapshot) error {
+func (s *Scorch) persistSnapshot(snapshot *IndexSnapshot) (err error) {
 	// start a write transaction
 	tx, err := s.rootBolt.Begin(true)
 	if err != nil {
 		return err
 	}
-	// defer fsync of the rootbolt
+	// defer sync or rollback
```

Comment:
hmmn, i really dont like this change, the pattern seemed like a clear, commit or rollback, and if you commit sync.  now it seems quite messy.  can you articulate what was wrong with the old approach?